.github/workflows/bundle-analysis.yml

```diff
@@ -0,0 +1,72 @@
+name: Bundle Size Analysis
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - 'lib/**'
```

Comment:
I think we need to trigger this, only when new dependencies are added or removed. So, I think only the `package.json` and `bun.lockb` should be there in paths.

cc: @seveibar

Reply:
done @imrishabh18